src/Parsers/Access/ASTCreateRowPolicyQuery.cpp

```diff
@@ -124,7 +124,21 @@ String ASTCreateRowPolicyQuery::getID(char) const
 
 ASTPtr ASTCreateRowPolicyQuery::clone() const
 {
-    return std::make_shared<ASTCreateRowPolicyQuery>(*this);
+    auto res = std::make_shared<ASTCreateRowPolicyQuery>(*this);
+
+    if (names)
+        res->names = std::static_pointer_cast<ASTRowPolicyNames>(names->clone());
+
+    if (roles)
+        res->roles = std::static_pointer_cast<ASTRolesOrUsersSet>(roles->clone());
+
+    for (auto & [_, res_filter] : res->filters)
+    {
+        if (res_filter)
+            res_filter = res_filter->clone();
```

Comment:
I didnt get why we are looping through `res->filters` ? maybe you want to loop `filters` and add to `res->filters`. 
Also do we not want to add `RowPolicyFilterType` to `res->filters`?

Reply:
`res->filters` is a vector which is already copied from `filters`, see

```
auto res = std::make_shared<ASTCreateRowPolicyQuery>(*this);
```

at the beginning of this function. So `res_filters` is already initialized before that loop because the copy-constructor of `ASTCreateRowPolicyQuery` has already done that at this point. However the copy constructor initialized `res_filters` with copying of smart pointers instead of calling `clone()` on them. But `ASTCreateRowPolicyQuery::clone()` should be implemented as a deep copy, not as a shallow copy. That's why I wrote that loop to transform a shallow copy into a deep copy. I think I need to write a comment in the code about that

Reply:
got it, thanks. As you suggested comment would be helpful.